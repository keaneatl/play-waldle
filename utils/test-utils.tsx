import { ReactNode, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import Theme from "../styles/theme";

type Props = {
  children: ReactNode;
};

function AllTheProviders({ children }: Props) {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
