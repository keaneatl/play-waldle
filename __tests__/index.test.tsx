import { render, screen } from "../utils/test-utils";
import Home from "../pages/index";
import "@testing-library/jest-dom";

test("it renders the right heading for the day", () => {
  render(<Home />);

  const heading = screen.getByRole("heading", {
    name: /waldle #1/i, // change to be dynamic
  });

  expect(heading).toBeInTheDocument();
});

test("it renders the poster", () => {
  render(<Home />);

  const displayImage = screen.getByAltText(
    /prehisoria - a collaboration of 31 pixeljoint artists/i
  );
  expect(displayImage).toMatchSnapshot();
});
