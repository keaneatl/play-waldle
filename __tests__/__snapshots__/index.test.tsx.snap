// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the poster 1`] = `
<img
  alt="PrehISOria - a collaboration of 31 PixelJoint artists"
  data-nimg="fill"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
/>
`;
